src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -996,6 +997,18 @@ internal override void ForceComplete(SourceLocation locationOpt, CancellationTok
             base.ForceComplete(locationOpt, cancellationToken);
         }
 
+        internal override bool IsDefinedInSourceTree(
+            SyntaxTree tree,
+            TextSpan? definedWithinSpan,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // Since only the declaring (and not the implementing) part of a partial method appears in the member
+            // list, we need to ensure we complete the implementation part when needed.
+            return
+                base.IsDefinedInSourceTree(tree, definedWithinSpan, cancellationToken) ||
+                this.SourcePartialImplementation?.IsDefinedInSourceTree(tree, definedWithinSpan, cancellationToken) == true;
```

Comment:
>this.SourcePartialImplementation?.IsDefinedInSourceTree(tree, definedWithinSpan, cancellationToken) == true; [](start = 16, length = 108)

Since we added this special case here, do we still need a an equivalent special case in CSharpCompilation.IsDefinedOrImplementedInSourceTree?

Reply:
No, we should be able to remove that. Will update.